### PR TITLE
Be able to inject middlewares from configuration

### PR DESCRIPTION
**Pull request by dgaubert**
For CRer, the following list are the highlighted commits to take a 👁:
- https://github.com/CartoDB/Windshaft-cartodb/pull/1126/commits/9cfaf6eefc635ae78748fb8fdcf310de0bf0664b
- https://github.com/CartoDB/Windshaft-cartodb/pull/1126/commits/3220e3de319036488c4e5f75e765f7f7663f4a4b
- https://github.com/CartoDB/Windshaft-cartodb/pull/1126/commits/dd06de2632661e19d64c9fbc2be0ba1a8059f54c
- https://github.com/CartoDB/Windshaft-cartodb/pull/1126/commits/a684bead928278add05c3fd575a134ad4e0f33ce
- https://github.com/CartoDB/Windshaft-cartodb/pull/1126/commits/93edf07da87ab8746ef9116a13ac3e46ba3dc45e

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -27,9 +27,9 @@ var argv = require('yargs')
     .usage('Usage: $0 <environment> [options]')
     .help('h')
     .example(
-    '$0 production -c /etc/sql-api/config.js',
-    'start server in production environment with /etc/sql-api/config.js as config file'
-)
+        '$0 production -c /etc/sql-api/config.js',
```
Comment (Algunenano) [inline app.js]: Consider adding the `node` command as you've done in the sql api command.

File: lib/cartodb/api/api-router.js [hunk 2]
```diff
@@ -196,16 +196,22 @@ module.exports = class ApiRouter {
...
-    register (app) {
+    route (app, routes) {
         // FIXME: we need a better way to reset cache while running tests
         if (process.env.NODE_ENV === 'test') {
             app.layergroupAffectedTablesCache = this.layergroupAffectedTablesCache;
         }
 
-        Object.keys(this.serverOptions.routes).forEach(apiVersion => {
-            const routes = this.serverOptions.routes[apiVersion];
+        // TODO: remove it before releasing next major version
```
Comment (Algunenano) [inline lib/cartodb/api/api-router.js]: I don't like this TODOs, either do it or keep the backwards compatibility please.

File: app.js (hunk 1 above)
Comment (dgaubert) [reply app.js]: 👌

File: lib/cartodb/api/api-router.js (hunk 2 above)
Comment (dgaubert) [reply lib/cartodb/api/api-router.js]: Neither do I to be honest. I'm going to think of another workaround.

**Pull request merged**